crates/aptos-in-memory-cache/src/lib.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use std::hash::Hash;
+
+pub mod caches;
+
+/// A trait for a cache that can be used to store key-value pairs.
+pub trait Cache<K, V>
```

Comment:
do we want this to support Send and Sync? The example in your PR indicates that S3Fifo is send and send but it's hidden behind the trait... for users, they need to wrap with Arc<Mutex<>>